presto-raptor/src/main/java/com/facebook/presto/raptor/filesystem/HdfsOrcDataEnvironment.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor.filesystem;
+
+import com.facebook.presto.orc.OrcDataSink;
+import com.facebook.presto.orc.OrcDataSource;
+import com.facebook.presto.orc.OrcDataSourceId;
+import com.facebook.presto.orc.OutputStreamOrcDataSink;
+import com.facebook.presto.raptor.storage.OrcDataEnvironment;
+import com.facebook.presto.raptor.storage.ReaderAttributes;
+import com.facebook.presto.spi.PrestoException;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+
+import static com.facebook.presto.raptor.RaptorErrorCode.RAPTOR_FILE_SYSTEM_ERROR;
+
+public class HdfsOrcDataEnvironment
+        implements OrcDataEnvironment
+{
+    private final FileSystem fileSystem;
+
+    @Inject
+    public HdfsOrcDataEnvironment(Path baseLocation, RaptorRemoteFileSystemConfiguration configuration)
+    {
+        try {
+            this.fileSystem = baseLocation.getFileSystem(configuration.getConfiguration());
+        }
+        catch (IOException e) {
+            throw new PrestoException(RAPTOR_FILE_SYSTEM_ERROR, "Raptor cannot create HDFS file system", e);
+        }
+    }
+
+    @Override
+    public FileSystem getFileSystem()
```

Comment:
Create another interface for `OrcDataEnvironment` so we can pass the query info and log it:
```java
public FileSystem getFileSystem(HdfsContext context)
```
The implementation of it for this class could be
```java
public FileSystem getFileSystem(HdfsContext context)
{
    return baseLocation.getFileSystem(configuration.getConfiguration(context));
}